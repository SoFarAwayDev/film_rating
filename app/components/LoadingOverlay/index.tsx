import React from 'react';
import styled from 'styles/styled-components';
import { colors } from 'styles/vars';
import CircleLoader from 'react-spinners/CircleLoader';

const Overlay = styled.div`
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0,0,0,0.5);
  z-index: 1000;
`;

export default function LoadingOverlay() {
  return (
    <Overlay>
      <CircleLoader
        size={150}
        color={colors.blue}
        loading={true}
      />
    </Overlay>
  );
}
