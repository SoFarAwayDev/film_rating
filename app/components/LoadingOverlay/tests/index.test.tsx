import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import LoadingOverlay from '../index';

describe('<LoadingOverlay />', () => {
  it('should match the snapshot', () => {
    const { container } = render(<LoadingOverlay />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
