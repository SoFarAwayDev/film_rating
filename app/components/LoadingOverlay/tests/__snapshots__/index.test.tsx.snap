// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingOverlay /> should match the snapshot 1`] = `
.c0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0,0,0,0.5);
  z-index: 1000;
}

<div
  class="c0"
>
  <div
    class="css-jv9gnm"
  >
    <div
      class="css-1c8n639"
    />
    <div
      class="css-grcojm"
    />
    <div
      class="css-11zj0bp"
    />
    <div
      class="css-13z2ea4"
    />
    <div
      class="css-1yvrs17"
    />
  </div>
</div>
`;
